### ISSUE TITLE
Improvement in "Could not connect" error

### ISSUE THREAD

**Issue by wbt** [trufflesuite/truffle#1664]
## TL;DR
This proposes making an error message more helpful with additional context information.

## Issue
I get the "Could not connect to your Ethereum client" error frequently, especially in load testing, regardless of Ethereum client being used.  I assume that what's happening (also in [issue 1461](https://github.com/trufflesuite/truffle/issues/1461)) is that the client is just getting overloaded with too many pending requests, and stops responding to new ones.  

If the issue were **not** an overloaded server that had just successfully responded to several transactions, and really was a connectivity issue, it would be helpful to know what configuration information was being read and used. 

## Steps to Reproduce
Set up a basic Truffle project to work with Ganache, don't start Ganache, and run `truffle migrate --reset`.

## Actual Results
Error message: 
>Could not connect to your Ethereum client. Please check that your Ethereum client:
    - is running
    - is accepting RPC connections (i.e., "--rpc" option is used in geth)
    - is accessible over the network
    - is properly configured in your Truffle configuration file (truffle.js)

## Expected Results
The same error, but with more specific information about where it was trying to connect to.  
>Could not connect to your Ethereum client at http://127.0.0.1:7545 (Network ID: 5777).
Please check that your Ethereum client:
    - is running
    - is accepting RPC connections (i.e., "--rpc" option is used in geth)
    - is accessible over the network
    - is properly configured in your Truffle configuration file (truffle.js), especially if the server information in the first line of this error does not match what you expect.

I would only expect "Network ID" to be there if it had been specified. 

## Environment
* Operating System: Windows 10 pro
* Ethereum client: For this Issue, none
* Truffle version (`truffle version`): 4.1.15
* node version (`node --version`): 9.3.0
* npm version (`npm --version`):  6.5.0

**Comment by wbt** [comment]
Per your comment [here](https://github.com/truffle-box/metacoin-box/pull/7#issuecomment-452820812), that last line should probably also refer to truffle-config.js instead of truffle.js.

**Comment by eggplantzzz** [comment]
Yep, I changed that line about 2 days ago! It should be out in the next patch release.

**Comment by eggplantzzz** [comment]
@wbt Would you like to open a PR for this? Let me know if you want to tackle it, otherwise I'll take it on next week.

**Comment by wbt** [comment]
Go for it, and thanks!

**Comment by gnidan** [comment]
This was merged! And released! Closing. Thanks!

**Comment by bater53** [comment]
Great to close a comment without giving a resolution to the issue.  I am running Windows 10 pro and I am having the same issue.  Now you will see that the ip address is different that is because under ganache I have  tried three different ones with the same results starting with 127.0.0.1 then going to 127.19.0.1 and finally with 127.17.224.1  all with the exact same results.  The only thing that changed was the ip address.   See the results on the last one below:  
/mnt/c/pr0/block-ater-bootcamp$ truffle migrate

Compiling your contracts...
===========================
> Everything is up to date, there is nothing to compile.
> Something went wrong while attempting to connect to the network at http://127.17.224.1:7545. Check your network configuration.
ProviderError:
Could not connect to your Ethereum client with the following parameters:
    - host       > 127.17.224.1
    - port       > 7545
    - network_id > *
Please check that your Ethereum client:
    - is running
    - is accepting RPC connections (i.e., "--rpc" or "--http" option is used in geth)
    - is accessible over the network
    - is properly configured in your Truffle configuration file (truffle-config.js)

    at /home/bsater/.nvm/versions/node/v16.14.0/lib/node_modules/truffle/build/webpack:/packages/provider/wrapper.js:76:1
    at /home/bsater/.nvm/versions/node/v16.14.0/lib/node_modules/truffle/build/webpack:/packages/provider/wrapper.js:114:1
    at XMLHttpRequest.request.onreadystatechange (/mnt/c/pr0/block-ater-bootcamp/node_modules/web3-providers-http/src/index.js:110:13)
    at XMLHttpRequestEventTarget.dispatchEvent (/mnt/c/pr0/block-ater-bootcamp/node_modules/xhr2-cookies/xml-http-request-event-target.ts:44:13)
    at XMLHttpRequest._setReadyState (/mnt/c/pr0/block-ater-bootcamp/node_modules/xhr2-cookies/xml-http-request.ts:219:8)
    at XMLHttpRequest._onHttpRequestError (/mnt/c/pr0/block-ater-bootcamp/node_modules/xhr2-cookies/xml-http-request.ts:379:8)
    at ClientRequest.<anonymous> (/mnt/c/pr0/block-ater-bootcamp/node_modules/xhr2-cookies/xml-http-request.ts:266:37)
    at ClientRequest.emit (node:events:520:28)
    at Socket.socketErrorListener (node:_http_client:442:9)
    at Socket.emit (node:events:520:28)
    at emitErrorNT (node:internal/streams/destroy:157:8)
    at emitErrorCloseNT (node:internal/streams/destroy:122:3)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)
Truffle v5.5.18 (core: 5.5.18)
Node v16.14.0

It would be nice if when you close the thread you also put in the resolution.  Remember the software we are using is still beta and so is Web3.  Any changes that are made can have castrophic endings on any legacy software that is out there.

**Comment by wbt** [comment]
@bater53 the PR which closed this is autolinked above.  You can see exactly what changes to the code were added to close the issue under the ["Files Changed" tab](https://github.com/trufflesuite/truffle/pull/1678/files) on that page.
